models/concerns/searchable.rb

```diff
@@ -10,14 +10,6 @@ module Searchable
     after_update :update_indexed_document
     after_destroy :delete_document
 
-    def self.put_search_index_mapping(index=nil)
```

Comment:
In task_helpers.rb `create_index`, I did a lot of experimenting to see if I could remove the current code used for creating the mappings with the index, and replace it with creating an index and then immediately following it with the equivalent of this call which uses `put_mapping`.  As far as I could tell, this results in the same exact index mappings as reported by `curl http://es.edx:9200/_mapping`.

Although I could detect no differences, some tests would fail (e.g. `bin/rspec -fd spec/api/search_spec.rb:201`).  The only difference I detected was in the search results (e.g. `curl http://es.edx:9200/_search`), where I saw `"group_id": null` for one comment_thread document in the failed version, but none in the passing (current) version of the code.

Note that this was all tested using the current ES 0.9.  

When we pick this up again, we could try this again using the 1.x+ branch.  At this point, I do not know what to recommend for migrations.